t/cs.t

```diff
@@ -2056,3 +2056,301 @@ GET /c
+                ngx.say("failed to connect: ", err)
+                return
+            end
+        }
+    }
+
+    location = /s {
+        content_by_lua_block {
+            local server = require "resty.websocket.server"
+            local wb, err = server:new()
+            if not wb then
+                ngx.log(ngx.ERR, "failed to new websocket: ", err)
+                return ngx.exit(444)
+            end
+
+            ngx.log(ngx.INFO, string.format("host: <%s>", ngx.var.http_host))
+            ngx.log(ngx.INFO, "SSL server name: ", ngx.var.ssl_server_name)
+        }
+    }
+--- request
+GET /c
+--- error_log
+host: <client.test>
+SSL server name: test.com
+
+
+
+=== TEST 34: overriding the Sec-WebSocket-Key header
+--- http_config eval: $::HttpConfig
+--- config
+    location = /c {
+        content_by_lua_block {
+            local client = require "resty.websocket.client"
+            local wb, err = client:new()
+            local uri = "ws://127.0.0.1:" .. ngx.var.server_port .. "/s"
+            local opts = {
+                key = "y7KXwBSpVrxtkR0O+bQt+Q==",
+            }
+            local ok, err = wb:connect(uri, opts)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+        }
+    }
+
+    location = /s {
+        content_by_lua_block {
+            local server = require "resty.websocket.server"
+            local wb, err = server:new()
+            if not wb then
+                ngx.log(ngx.ERR, "failed to new websocket: ", err)
+                return ngx.exit(444)
+            end
+
+            ngx.log(ngx.INFO, "key: ", ngx.var.http_sec_websocket_key)
+        }
+    }
+--- request
+GET /c
+--- error_log
+key: y7KXwBSpVrxtkR0O+bQt+Q==
+
+
+
+=== TEST 35: keeping the server response
+--- http_config eval: $::HttpConfig
+--- config
+    location = /c {
+        content_by_lua_block {
+            local client = require "resty.websocket.client"
+            local wb, err = client:new()
+            local uri = "ws://127.0.0.1:" .. ngx.var.server_port .. "/s"
+            local opts = {
+                keep_response = true,
+            }
+            local ok, err, res = wb:connect(uri, opts)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            if not res then
+                ngx.say("no response string")
+                return
+            end
+            ngx.say(res)
+        }
+    }
+
+    location = /s {
+        content_by_lua_block {
+            ngx.header["x-test"] = "test"
+            ngx.header["x-multi"] = { "one", "two" }
+
+            local server = require "resty.websocket.server"
+            local wb, err = server:new()
+            if not wb then
+                ngx.log(ngx.ERR, "failed to new websocket: ", err)
+                return ngx.exit(444)
+            end
+        }
+    }
+--- request
+GET /c
+--- response_body_unlike
```

Comment:
Can't we test this with `response_body_like` instead? As it stands this test is susceptible to false-positives.